plenum/server/node.py

```diff
@@ -2338,10 +2326,68 @@ def allLedgersCaughtUp(self):
             logger.info('{}{} does not need any more catchups'
                         .format(CATCH_UP_PREFIX, self),
                         extra={'cli': True})
+
+            self.selection_primaries_after_catchup()
+
+            # set view_change_in_progress to False
             self.no_more_catchups_needed()
-            # select primaries after pool ledger caughtup
-            if not self.view_change_in_progress:
-                self.select_primaries()
+
+    def selection_primaries_after_catchup(self):
```

Comment:
It would be great to move all primary selection logic into a separate class.

Reply:
Yes, we can do that. But we need to agree what is needed to be done at the end of catchup